File: Data Structures & Algorithms/binary-search/submission-2.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number}
     */
    search(nums, target) {
        let low = 0;
        let high = nums.length - 1
        while (low <= high){
            let middle = Math.floor(low + (high - low) / 2);
            let value = nums[middle]
            if (value === target) return middle;
            else if (value > target){
                high = middle - 1
            } else {
                low = middle + 1
            }
        }
        return -1
    }
}

File: Data Structures & Algorithms/duplicate-integer/submission-2.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        thisDict = dict()
        if len(nums) < 1:
            return False
        for i,v in enumerate(nums):
            if v in thisDict.keys():
                thisDict[v] += 1
                return True
            else:
                thisDict[v] = 1
        return False
         